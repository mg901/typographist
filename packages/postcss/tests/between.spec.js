const run = require('./run');

describe('base', () => {
  test('with breakpoints name', async () => {
    const source = `
    .test {
      @between(tablet, desktop) {
          color: hotpink;
      }
    }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with pixels', async () => {
    const source = `
    .test {
      @between(1000px, 2000px) {
        color: hotpink;
      }
    }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with em', async () => {
    const source = `
              .test {
                @between(30em, 50em) {
                  color: hotpink;
                }
              }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `landscape`', async () => {
    const source = `
                .test {
                  @between(30em, 50em):landscape {
                    color: hotpink;
                  }
                }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `portrait`', async () => {
    const source = `
            .test {
              @between(30em, 50em):portrait {
                color: hotpink;
              }
            }`;

    expect(await run(source)).toMatchSnapshot();
  });
});
