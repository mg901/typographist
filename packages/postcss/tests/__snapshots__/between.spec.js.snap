// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base with breakpoints name 1`] = `
@media (min-width: 48em) and (max-width: 74.99875em) {
  .test {
    color: hotpink;
  }
}
`;

exports[`base with em 1`] = `
@media (min-width: 30em) and (max-width: 49.99875em) {
  .test {
    color: hotpink;
  }
}
`;

exports[`base with orientation \`landscape\` 1`] = `
@media (min-width: 30em) and (max-width: 49.99875em) and (orientation: landscape) {
  .test {
    color: hotpink;
  }
}
`;

exports[`base with orientation \`portrait\` 1`] = `
@media (min-width: 30em) and (max-width: 49.99875em) and (orientation: portrait) {
  .test {
    color: hotpink;
  }
}
`;

exports[`base with pixels 1`] = `
@media (min-width: 62.5em) and (max-width: 124.99875em) {
  .test {
    color: hotpink;
  }
}
`;
